bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -450,9 +778,49 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
          * Returns an array of layers added to the service for example via #addLayer()
          * @return {Mixed[]/Oskari.mapframework.domain.WmsLayer[]/Oskari.mapframework.domain.WfsLayer[]/Oskari.mapframework.domain.VectorLayer[]/Object[]}
          */
-        getAllLayers: function () {
+        getAllLayers: function() {
             return this._loadedLayersList;
         },
+
+        /**
+         * @method getAllLayerGroups
+         * Returns an array of layer groups added to the service
+         * @param {String|Integer} id if defined return only wanted group
+         * @return {Mixed[]/Oskari.mapframework.domain.WmsLayer[]/Oskari.mapframework.domain.WfsLayer[]/Oskari.mapframework.domain.VectorLayer[]/Object[]}
+         */
+        getAllLayerGroups: function(id) {
+            if (id) {
+                var filterFunction = function(group) {
+                    return group.id == id;
+                };
+                var group = this._layerGroups.filter(filterFunction)[0];
```

Comment:
The Java-person inside me screams ArrayIndexOutOfBounds, but taking the first item of an empty array doesn't seem to bother JS too much so I guess this is ok.